mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassin.java

```diff
@@ -41,21 +42,25 @@ public SpamAssassin(MetricFactory metricFactory, SpamAssassinConfiguration spamA
         this.spamAssassinConfiguration = spamAssassinConfiguration;
     }
 
-    public void learnSpam(List<InputStream> messages, Username username) {
+    public Mono<Void> learnSpam(List<InputStream> messages, Username username) {
         if (spamAssassinConfiguration.isEnable()) {
             Host host = spamAssassinConfiguration.getHost().get();
             SpamAssassinInvoker invoker = new SpamAssassinInvoker(metricFactory, host.getHostName(), host.getPort());
-            messages
-                .forEach(Throwing.consumer(message -> invoker.learnAsSpam(message, username)));
+            return Flux.fromIterable(messages)
+                .flatMap(message -> invoker.learnAsSpam(message, username))
+                .then();
         }
+        return Mono.empty();
     }
 
-    public void learnHam(List<InputStream> messages, Username username) {
+    public Mono<Void> learnHam(List<InputStream> messages, Username username) {
         if (spamAssassinConfiguration.isEnable()) {
             Host host = spamAssassinConfiguration.getHost().get();
             SpamAssassinInvoker invoker = new SpamAssassinInvoker(metricFactory, host.getHostName(), host.getPort());
-            messages
-                .forEach(Throwing.consumer(message -> invoker.learnAsHam(message, username)));
```

Comment:
Already done in #3281

Reply:
first merged rebase the other one

Reply:
There will be a conflict but it's not exactly the same thing as here the SpamAssassinInvoker return a Mono, hence the flatMap.
The merge will be pretty simple. (drop a Mono.callable in your code when it call the invoker)

Reply:
not really, I do not return a `Mono` in my `learn*`.

But if you find it simple, please, do it :)

Reply:
it will have to be done when the conflict arise. Because i don't think rebasing one PR on the other will help it to be merged :p